polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -2142,8 +2202,17 @@ MLIRScanner::EmitSYCLOps(const clang::Expr *Expr,
 
       std::string name;
       MLIRScanner::getMangledFuncName(name, Func, Glob.getCGM());
-      Op = builder.create<mlir::sycl::SYCLCallOp>(
-          loc, OptRetType, OptFuncType, Func->getNameAsString(), name, Args);
+      if (OptFuncType) {
```

Comment:
I guess could add this comment here: 

// Attempt to create a SYCL method call first, if that fails create a generic SYCLCallOp.

P.S.: I am wondering what is special about a sycl.call operation now that we have more specific operations for SYCL member functions. Do we need sycl.call at all or can we just use a regular MLIR call operation? 

If we want to create specialized operations for *every* method of a SYCL class then I don't think we need to also have a sycl.call operation, conversely if we want to specialize a subset of all the possible SYCL methods perhaps a sycl.call operation still make sense.

Reply:
We were planning on specializing just a subset of the methods. Also, there are plenty of arithmetic functions which we are not planning on specializing.